S3/S3.py

```diff
@@ -1725,6 +1726,10 @@ def send_file(self, request, stream, labels, buffer = '', throttle = 0,
             sha256_hash = checksum_sha256_file(stream, offset, size_total)
         request.body = sha256_hash
 
+        # Provide the checksum with teh request. This is important for buckets that have 
+        # Object Lock enabled.
+        headers['x-amz-checksum-sha256'] = base64.b64encode(sha256_hash.digest()).decode()
```

Comment:
As someone could use s3cmd as library this line can override the header potentially comes with request argument's headers attribute. I think it's better to check if it currently has values:
```python
if not headers.get("x-amz-checksum-sha256"):
    headers['x-amz-checksum-sha256'] = sha256_hash_to_base64(sha256_hash)
```

Also in this module encodestring (which imported above as an alias of encodebytes when available) is the function used for base64 encoding.
I think you can use:
```python
sha256_hash_digest_b64 = encodestring(sha256_hash_digest).strip().decode()
```
to align with other usecases.